tests/CMakeLists.txt

```diff
@@ -108,7 +125,11 @@ pybind11_add_module(pybind11_tests pybind11_tests.cpp
 pybind11_enable_warnings(pybind11_tests)
 
 if(EIGEN3_FOUND)
-  target_include_directories(pybind11_tests PRIVATE ${EIGEN3_INCLUDE_DIR})
+  if (PYBIND11_EIGEN_VIA_TARGET)
+    target_link_libraries(pybind11_tests PRIVATE Eigen3::Eigen)
```

Comment:
Perhaps we can just make the outside-the-tree build require cmake 3.0 and be done with it.